src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -463,28 +522,54 @@ bool BAMUtils::hasValidFastaIndex(const GUrl& fastaUrl) {
     }
 }
 
-void BAMUtils::createBamIndex(const GUrl& bamUrl, U2OpStatus& os) {
-    const QByteArray& bamFileName = bamUrl.getURLString().toLocal8Bit();
+/**
+ * Builds and saves index for BAM file. Returns 0 if the index was created correctly.
+ * Exact copy of 'bam_index_build2' with a correct unicode file names support.
+ */
+static int bam_index_build_unicode(const QString& bamFileName) {
+    FILE* bFile = BAMUtils::openFile(bamFileName, "rb");
+    CHECK(bFile != nullptr, -1);
+    bamFile fp = bam_dopen(fileno(bFile), "rb");
+    if (fp == nullptr) {
+        closeFileIfOpen(bFile);
+        fprintf(stderr, "[bam_index_build2] fail to open the BAM file.\n");
+        return -1;
+    }
+    fp->owned_file = 1;
+    bam_index_t* idx = bam_index_core(fp);
+    bam_close(fp);
+    if (idx == nullptr) {
+        fprintf(stderr, "[bam_index_build2] fail to index the BAM file.\n");
+        return -1;
+    }
+    FILE* fpidx = BAMUtils::openFile(bamFileName + ".bai", "wb");
+    if (fpidx == nullptr) {
+        fprintf(stderr, "[bam_index_build2] fail to create the index file.\n");
+        return -1;
+    }
+    bam_index_save(idx, fpidx);
+    bam_index_destroy(idx);
+    fclose(fpidx);
```

Comment:
Has a return value, which could be an error, need to check it

Reply:
Same here: this is an original behavior of samtools.